cami.c

```diff
@@ -438,18 +484,35 @@ struct ami_session *ami_connect(const char *hostname, int port, void (*callback)
 		pthread_mutexattr_destroy(&attr);
 	}
 
-	if (pthread_create(&ami->ami_thread, NULL, ami_loop, ami)) {
-		ami_error(ami, "Unable to create AMI thread: %s\n", strerror(errno));
-		ami_cleanup(ami);
-		goto cleanup;
+	{
+		pthread_attr_t attr;
+		pthread_attr_init(&attr);
+		pthread_attr_setstacksize(&attr, 2 * 1024 * 1024);
+		ret = pthread_create(&ami->ami_thread, &attr, ami_loop, ami);
+		pthread_attr_destroy(&attr);
+		if (ret) {
+			ami_error(ami, "Unable to create AMI thread: %s\n", strerror(errno));
+			ami_cleanup(ami);
+			goto cleanup;
+		}
 	}
-	if (pthread_create(&ami->dispatch_thread, NULL, ami_event_dispatch, ami)) {
-		ami_error(ami, "Unable to create dispatch thread: %s\n", strerror(errno));
-		ami_cleanup(ami);
-		goto cleanup;
+
+	{
+		pthread_attr_t attr;
+		pthread_attr_init(&attr);
+		pthread_attr_setstacksize(&attr, 2 * 1024 * 1024);
+		ret = pthread_create(&ami->dispatch_thread, &attr, ami_event_dispatch, ami);
+		pthread_attr_destroy(&attr);
+		if (ret) {
+			ami_error(ami, "Unable to create dispatch thread: %s\n", strerror(errno));
+			ami_cleanup(ami);
+			goto cleanup;
+		}
 	}
 
 	pthread_mutex_unlock(&ami->ami_read_lock);
+	ami->debugfd = -1;
```

Comment:
Add a comment this is to restore from the default debug level and user can set per session afterwards